lib/context.js

```diff
@@ -116,6 +110,13 @@ class ResponseContext {
         post_logout_redirect_uri: returnURL,
         id_token_hint: req.openid.tokens,
       });
+
+      if (typeof req.session.destroy === 'function') {
```

Comment:
Great catch here! But can we move this and the redirect outside of the `try` block? It's only `client.endSessionUrl` that will throw an error so having the rest in there makes it a little less clear.

Reply:
Hmm. If `client.endSessionUrl` must come first and the redirect is dependent on it's result (`url`), the only way I can see it to achieve is using a `let`:
```js
const client = this._req.openid.client;
let url;
try {
  url = client.endSessionUrl({
    post_logout_redirect_uri: returnURL,
    id_token_hint: req.openid.tokens,
  });
} catch(err) {
  next(err);
}

if (typeof req.session.destroy === 'function') {
  req.session.destroy();
} else {
  req.session = null;
}
res.redirect(url);
```

Am I not seeing something, or you think this is the way to go?

Reply:
I'm not wild about it either but I think it's more clear than packing all that work into a try block. If an error happens from this call, it's much more clear where it came from.